src/main/webapp/consentGroupDocuments/ConsentGroupDocuments.js

```diff
@@ -29,56 +31,64 @@ export const ConsentGroupDocuments = hh(class ConsentGroupDocuments extends Comp
   }
 
   componentDidMount() {
+    this._isMounted = true;
     this.getAttachedDocuments();
     this.getUseRestriction();
     this.loadOptions();
     this.getAssociatedProjects();
   }
 
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
   loadOptions() {
     let documentOptions = [];
     CONSENT_DOCUMENTS.forEach(type => {
       documentOptions.push({ value: type, label: type });
     });
-    this.setState({ documentOptions: documentOptions });
+    if (this._isMounted) {
+      this.setState({ documentOptions: documentOptions });
+    }
   };
 
   getAttachedDocuments = () => {
     DocumentHandler.attachedDocuments(this.props.consentKey).then(resp => {
       User.getUserSession().then(user => {
-        this.setState(prev => {
-            prev.documents = JSON.parse(resp.data.documents);
-            prev.user = user.data;
-            return prev;
-          }, () => {
-            this.props.updateDocumentsStatus({ attachmentsApproved: resp.data.attachmentsApproved })
-          }
-        );
+        if (this._isMounted) {
+          this.setState(prev => {
```

Comment:
here too

Reply:
In this case is used the callback of the `setState`.